controllers/argocd/role.go

```diff
@@ -103,8 +103,13 @@ func (r *ReconcileArgoCD) reconcileRole(name string, policyRules []v1.PolicyRule
 
 	// create policy rules for each namespace
 	for _, namespace := range r.ManagedNamespaces.Items {
-		// Skip terminating namespaces.
+		// If encountering a terminating namespace remove managed-by label from it and skip reconciliation - This should trigger
```

Comment:
It _should_ trigger or it _will_ trigger?

Can we back this up by a test?

Reply:
@jannfis 
it will trigger
Added a test case, PTAL 
(Only checking for synced state of app because the app likely runs privileged containers which will fail to come up on OCP and will stay in progressing/degraded state. Sync state anyway should confirm that the resources were at least deployed in the target namespace, which should be enough to address the original bug)